src/core/sock/sockinfo.h

```diff
@@ -159,7 +159,7 @@ class sockinfo : public socket_fd_api,
                  public pkt_sndr_source,
                  public wakeup_pipe {
 public:
-    sockinfo(int fd, int domain, bool use_ring_locks);
+    sockinfo(int fd, int domain, bool use_ring_locks, uint8_t sock_type);
```

Comment:
You can use `PROTO_TCP == s->get_protocol()`

Reply:
done.